erpnext/buying/report/quoted_item_comparison/quoted_item_comparison.py

```diff
@@ -2,103 +2,123 @@
 # For license information, please see license.txt
 
 from __future__ import unicode_literals
-from erpnext.setup.utils import get_exchange_rate
-from frappe.utils import flt, cint
 import frappe
+from frappe.utils import flt, cint
+from frappe import _
+from collections import defaultdict
+from erpnext.setup.utils import get_exchange_rate
 
 def execute(filters=None):
-	qty_list = get_quantity_list(filters.item)
-	data = get_quote_list(filters.item, qty_list)
-	columns = get_columns(qty_list)
+	conditions = get_conditions(filters)
+	data = get_data(filters, conditions)
+	columns = get_columns()
 	return columns, data
-	
-def get_quote_list(item, qty_list):
-	out = []
+
+def get_data(filters, conditions):
+	out, suppliers = [], []
+	item = filters.get("item")
+
 	if not item:
 		return []
 
-	suppliers = []
-	price_data = []
 	company_currency = frappe.db.get_default("currency")
-	float_precision = cint(frappe.db.get_default("float_precision")) or 2 
-	# Get the list of suppliers
-	for root in frappe.db.sql("""select parent, qty, rate from `tabSupplier Quotation Item`
-		where item_code=%s and docstatus < 2""", item, as_dict=1):
-		for splr in frappe.db.sql("""select supplier from `tabSupplier Quotation`
-			where name =%s and docstatus < 2""", root.parent, as_dict=1):
-			ip = frappe._dict({
-				"supplier": splr.supplier,
-				"qty": root.qty,
-				"parent": root.parent,
-				"rate": root.rate
-			})
-			price_data.append(ip)
-			suppliers.append(splr.supplier)
-
-	#Add a row for each supplier
-	for root in set(suppliers):
-		supplier_currency = frappe.db.get_value("Supplier", root, "default_currency")
+	float_precision = cint(frappe.db.get_default("float_precision")) or 2
+
+	supplier_quotation_data = frappe.db.sql("""SELECT
+		sqi.parent, sqi.qty, sqi.rate, sqi.uom, sqi.request_for_quotation,
+		sq.supplier
+		FROM
+			`tabSupplier Quotation Item` sqi,
+			`tabSupplier Quotation` sq
+		WHERE
+			sqi.item_code = '{0}'
```

Comment:
If item has special character then this may break, either do frappe.db.escape(item) or %s (see below query) 

```
	supplier_quotation_data = frappe.db.sql("""SELECT
		sqi.parent, sqi.qty, sqi.rate, sqi.uom, sqi.request_for_quotation,
		sq.supplier
		FROM
			`tabSupplier Quotation Item` sqi,
			`tabSupplier Quotation` sq
		WHERE
			sqi.item_code = %s
			AND sqi.parent = sq.name
			AND sqi.docstatus < 2
			AND sq.company = '{1}'
			AND sq.status != 'Expired'
			{2}""".format(filters.get("company"), conditions), item, as_dict=1)
```